src/analysis/retail/evoker/preservation/modules/talents/TimeOfNeed.tsx

```diff
@@ -0,0 +1,234 @@
+import { formatNumber } from 'common/format';
+import fetchWcl from 'common/fetchWclApi';
+import { WCLEventsResponse } from 'common/WCL_TYPES';
+import { Trans } from '@lingui/macro';
+import Combatants from 'parser/shared/modules/Combatants';
+
+//All possible results from a Time of Need proc
+enum Result {
+  Proc = 'Saved by Verdant Embrace',
+  Long = 'Saved by all the healing',
+  Not = 'Was not in danger of death',
+  Dead = 'Died anyway',
+  Bug = 'Time of Need fizzled out',
+  Unparsed = "This event wasn't reviewed",
+}
+
+interface tonEvent {
+  summon: SummonEvent;
+  verdantEmbrace: HealEvent | null;
+  livingFlames: HealEvent[];
+  livingFlameTotalHeal: number;
+  livingFlameTotalOverheal: number;
+  target: number;
+  result: Result;
+}
+
+class TimeOfNeed extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  protected combatants!: Combatants;
+  spawns: tonEvent[] = [];
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.hasTalent(TALENTS_EVOKER.TIME_OF_NEED_TALENT);
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.summon.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.TIME_OF_NEED_SUMMON),
+      this.onSpawn,
+    );
+  }
+
+  onSpawn(event: SummonEvent) {
+    const HealEvents = getTimeOfNeedHealing(event);
+
+    const currentEvent: tonEvent = {
+      summon: event,
+      verdantEmbrace: null,
+      livingFlames: [],
+      livingFlameTotalHeal: 0,
+      livingFlameTotalOverheal: 0,
+      target: 0,
+      result: Result.Not,
+    };
+
+    HealEvents.forEach((heal) => {
+      if (heal.ability.guid === SPELLS.VERDANT_EMBRACE_HEAL.id) {
+        currentEvent.verdantEmbrace = heal;
+        currentEvent.target = heal.targetID;
+      } else if (heal.ability.guid === SPELLS.TIME_OF_NEED_LIVING_FLAME.id) {
+        currentEvent.livingFlames.push(heal);
+        currentEvent.livingFlameTotalHeal += heal.amount + (heal.absorbed || 0);
+        currentEvent.livingFlameTotalOverheal += heal.overheal ?? 0;
+      }
+    });
+
+    this.spawns.push(currentEvent);
+  }
+
+  parseDamageTaken(tonSpawn: tonEvent, eventList: WCLEventsResponse) {
+    //Check that ToN actually did any healing or if it bugged out
+    if (tonSpawn.verdantEmbrace === null && tonSpawn.livingFlames.length === 0) {
+      return Result.Bug;
+    }
+    //If there is no damage events, then Time of Need didn't save the player from anything
+    if (eventList.events.length === 0) {
+      return Result.Not;
+    }
+    const hits = eventList.events as DamageEvent[];
+    //Iterate all the damage taken events during the time window
+    for (const hit of hits) {
+      //Check that the info from the damage event is valid
+      if (hit.hitPoints !== undefined && hit.targetID === tonSpawn.target) {
+        //If the hitpoints after the hit are equal to 0, the player died anyway
+        if (hit.hitPoints === 0) {
```

Comment:
```suggestion
        if (!hit.hitPoints) {
```
not gonna do it for every case, but prefer doing `!a.property` instead of `a.property === 0`